src/scaffoldmaker/utils/eftfactory_tricubichermite.py

```diff
@@ -819,6 +819,180 @@ def createEftWedgeXi1Zero(self):
+
+        # remap parameters on xi3 = 1 before collapsing nodes
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS1, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS2, [])])
+        remapEftNodeValueLabel(eft, [7], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS1, [])])
+
+        # remap parameters on xi2 = 0 before collapsing nodes
+        remapEftNodeValueLabel(eft, [1, 2], Node.VALUE_LABEL_D_DS1, [])
+        remapEftNodeValueLabel(eft, [1], Node.VALUE_LABEL_D_DS2, [(Node.VALUE_LABEL_D_DS1, [1])])
+
+        # zero cross derivative parameters
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS2DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [1, 1, 2, 3, 4, 4, 4, 4]
+        remapEftLocalNodes(eft, 4, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftTetrahedronCollapseXi1Xi2AtXi3OneXi1AtXi2Zero:  Failed to validate eft'
+        return eft
+
+    def createEftTetrahedronCollapseXi1Xi2AtXi3OneXi1AtXi2One(self):
+        '''
+        Create a tricubic hermite element field for a tetrahedron element, where xi1 and xi2 are collapsed on xi3 = 1
+        , and then, xi1 for nodes on both xi2 = 1 and xi3 = 0 is collapsed.
+        :return: Element field template
+        '''
+        eft = self.createEftBasic()
+        setEftScaleFactorIds(eft, [1], [])
+
+        # remap parameters on xi3 = 1 before collapsing nodes
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS1, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D_DS2, [])
+        remapEftNodeValueLabel(eft, [7, 8], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS2, [1])])
+        remapEftNodeValueLabel(eft, [5], Node.VALUE_LABEL_D_DS3, [(Node.VALUE_LABEL_D_DS1, [])])
+
+        # remap parameters on xi2 = 1 before collapsing nodes
+        remapEftNodeValueLabel(eft, [3, 4], Node.VALUE_LABEL_D_DS1, [])
+        remapEftNodeValueLabel(eft, [3], Node.VALUE_LABEL_D_DS2, [(Node.VALUE_LABEL_D_DS1, [])])
+
+        # zero cross derivative parameters
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS2, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS1DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D2_DS2DS3, [])
+        remapEftNodeValueLabel(eft, [5, 6, 7, 8], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [1, 2, 3, 3, 4, 4, 4, 4]
+        remapEftLocalNodes(eft, 4, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftTetrahedronCollapseXi1Xi2AtXi3OneXi1AtXi2One:  Failed to validate eft'
+        return eft
```

Comment:
Similar to the wedge, need a naming scheme to manage all these permutations, suggest for all tetrahedra collapsed in xi1xi2:
``createEftTetrahedronCollapseXi1Xi2Quadrant(self, peakNode, sideNode)``
where peakNode is the localNode where the whole of xi1, xi2 is collapsed on to (on either xi3=0 or xi3=1), and sideNode is the node at the other end of xi3 which has another node collapsed onto it.
e.g.
``eft = tricubicfactory.createEftTetrahedronCollapseXi1Xi2Quadrant(8, 2)`` would do the same as ``createEftTetrahedronCollapseXi1Xi2AtXi3OneXi1AtXi2Zero``

Reply:
So, now I replaced 3 functions for creating wedge elements which cover 12 cases.

Reply:
I defined one function named "createEftTetrahedronCollapseXi1Xi2QuadrantAtXi3One" which create four types of tetrahedron. Later, other cases can be added.